### PR TITLE
Update Router Templating API docs

### PR DESCRIPTION
**Pull request by ddunn2**
## Description
Update the `TemplateEngine` docs to provide more information regarding edge cases such as registering multiple different template engines to use the same extension. 

## Motivation and Context
Makes the API docs clearer for users.

### CODE REVIEW

File: Sources/Kitura/Router.swift [hunk 1]
```diff
@@ -126,14 +126,22 @@ public class Router {
...
+    /// that match the file extension it supports. You can change this default location using the `viewsPath` property.
     /// ### Usage Example: ###
     /// ```swift
     /// let router = Router()
+    /// //Register a template engine to a given router instance
     /// router.add(templateEngine: MyTemplateEngine())
+    /// //Register a template engine to a given router instance using a custom file extension
     /// router.add(templateEngine: MyOtherTemplateEngine(), forFileExtensions: ["html"], useDefaultFileExtension: false)
+    /// //Change the location a template engine will look for template files in any given router instance
+    /// router.viewsPath = "New_Custom_Location"
```
Comment (alexwishart) [inline Sources/Kitura/Router.swift]: This doesn't need to be here, it's already been covered in https://ibm-swift.github.io/Kitura/Classes/Router.html#/s:6Kitura6RouterC9viewsPathSSv

File: Sources/Kitura/Router.swift [hunk 2]
```diff
@@ -156,6 +164,9 @@ public class Router {
     /// let router = Router()
     /// router.setDefault(templateEngine: MyTemplateEngine())
     /// ```
+    /// If the template engine doesn't provide a default extension you can provide one using
+    /// `add(templateEngine:forFileExtensions:useDefaultFileExtension:)`. If a router instance doesn't
+    /// have a template engine registered that can render the given template file an error is thrown.
```
Comment (alexwishart) [inline Sources/Kitura/Router.swift]: Could you add in what the error is, then people can recognise that it's the "template engine missing" error and not something else.

File: Sources/Kitura/Router.swift [hunk 3]
```diff
@@ -126,14 +126,22 @@ public class Router {
 
     // MARK: Template Engine
 
-    /// Register a templating engine. The templating engine will handle files in the `viewsPath`
-    /// that match the file extension it supports.
+    /// A template engine allows rendering of documents using static templates.
```
Comment (alexwishart) [inline Sources/Kitura/Router.swift]: Convention in the other functions is that the first thing you see explicitly says what the specific function does e.g. "Set up a sub router to handle requests." With any extra explanation coming after. I think if you just move the first comment line from your code (i.e. "Register a template engine to a given router instance") to the top here it's fine 🙂

Comment (ddunn2) [reply Sources/Kitura/Router.swift]: I'll look into how to best phrase this as 'Register a templating engine' just felt out of place there. I like your idea of moving the comment up to the top, that could work with a little rewording :).

File: Sources/Kitura/Router.swift [hunk 4]
```diff
@@ -126,14 +126,22 @@ public class Router {
...
+    ///
+    /// By default the templating engine will handle files in the `./Views` directory
+    /// that match the file extension it supports. You can change this default location using the `viewsPath` property.
     /// ### Usage Example: ###
     /// ```swift
     /// let router = Router()
+    /// //Register a template engine to a given router instance
```
[the remark this answers is not in the document]
Comment (ddunn2) [reply Sources/Kitura/Router.swift]: I think having no comments is the 'old' convention. In the Codable routing API we have comments. I will however follow whichever convention, so will remove the comments.

[the remark this answers is not in the document]
Comment (alexwishart) [reply Sources/Kitura/Router.swift]: @helenmasters can you confirm r.e. comments/no comments? Maybe we need to create some documentation somewhere on how we want the APIs to look (although I think we'd started to use [Kitura-Cache](https://github.com/IBM-Swift/Kitura-Cache ) as an example?)

File: Sources/Kitura/Router.swift [hunk 5]
```diff
@@ -156,6 +161,10 @@ public class Router {
     /// let router = Router()
     /// router.setDefault(templateEngine: MyTemplateEngine())
     /// ```
+    /// If the template engine doesn't provide a default extension you can provide one using
+    /// `add(templateEngine:forFileExtensions:useDefaultFileExtension:)`. If a router instance doesn't
+    /// have a template engine registered that can render the given template file an
```
Comment (alexwishart) [inline Sources/Kitura/Router.swift]: "an" should be "a"

File: Sources/Kitura/Router.swift [hunk 6]
```diff
@@ -156,6 +161,10 @@ public class Router {
...
+    /// `No template engine defined for extension` error is thrown.
```
Comment (ianpartridge) [inline Sources/Kitura/Router.swift]: This is a `TemplatingError` I think - only use backticks for code.

**Comment by ddunn2** [comment]
Addressed review comments, will ask @helenmasters to review as well.

**Pull request merged**